### PR TITLE
bugfix/20334-win-width

### LINKED ISSUE #20334: Highcharts width exceeds container div on first load
When viewing a chart in a small screen, the chart overflows and gets hidden under the vertical scrollbar. This issue is reproducible when the vertical scrollbar is not present initially and appears when the chart is fully rendered.
image

Once a resize happens, chart redraws and the issue is fixed.
image

image
highcharts-container seems to be overflowing the container

This issue is actual only for small screens and works fine if scrollbar is not needed or if scrollbar is present without highcharts

#### Expected behaviour
Y axis labels should not overflow the container

#### Actual behaviour
Y axis labels seems to overflow the container

Internally in our projects, while using similar configuration, the tooltip is also overflowing outside the chart window
image

#### Live demo with steps to reproduce
https://jsfiddle.net/jijojames18/0rdf9sha/52/

#### Product version
Highstock v11.2.0

#### Affected browser(s)
All major browsers

### PR DESCRIPTION
**Pull request by KacperMadej**
Fixed #20334, chart width exceeded container width on Windows when scrollbars were present.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206237632708354

### CODE REVIEW

**Review by Xillez — APPROVED** [review]
LGTM 👍 Seems to work nicely

**Comment by TorsteinHonsi** [comment]
Thanks for the PR! 

Before we commit this - do we know why this broke in v11? I have problems reproducing this myself in BrowserStack, but I suspect it may have something to do with the `ResizeObserver` refactoring that we did.

For example, what happens if we remove the `chart.hasLoaded` condition here?

https://github.com/highcharts/highcharts/blob/5ebada8704aee914e06d7effe03187babd444650/ts/Core/Chart/Chart.ts#L1815

Will it magically redraw to make room for the scrollbars? If not, can we bisect the issue to a specific commit?

Maybe we can get away with a cheaper fix if we know exactly what change that broke it.

**Comment by KacperMadej** [comment]
@TorsteinHonsi 
Yes, this is related to the `ResizeObserver` refactor.

**Some context:**

I could recreate the previously "working" case only when there was an extra reflow running that was doing nothing (except maybe wasting time and resources) in almost all charts. The behavior was never perfect as in the code reported in the issue there is `chart.reflow` on the chart's `redraw` event ([more in the comment here](https://github.com/highcharts/highcharts/issues/20334#issuecomment-1924271703)). Moving to the `load` event would be an improvement but neither works now because in this [here](https://github.com/highcharts/highcharts/pull/18770/files#diff-eec054049d33d5ccd9ad7485a735dfeb5bda93fba3e975bdd69cdff552d400fbR1649) the container size is read and updated for the reference of the reflow function, so the initial reflow doesn't detect the change. Only `chart.continerBox` updates but nothing was adjusted.

`chart.getContainerBox();` is needed because when the `ResizeObserver` starts observing it runs the callback on activation (as expected by the spec - IMO explained better [here](https://stackoverflow.com/questions/60026223/does-resizeobserver-invoke-initially-on-page-load)), so I moved it earlier in the code execution - now the width could be registered early and an instance of Renderer gets the correct size on init.

A funny thing (maybe not if you start to question your sanity and decide to continue writing a comment the next day, but still), if you set up break-points just right you will or won't get the initial callback execution. When you observe the observer closely enough it behaves differently - without break-points the DOM is updated after the `ResizeObserver` starts observing.

**In conclusion:**

> what happens if we remove the `chart.hasLoaded` condition here?

When I am testing this, the first time `runReflow` is executed `chart.hasLoaded` is already `true`, so removing it will change nothing. In chart's `reflow` function setSize doesn't happen because `oldBox` is the same as the freshly obtained `containerBox`. This is the same behavior for the code with and without the fix, but with the fix the match between `oldBox` and `containerBox` truly represents the lack of change in size (and without the fix, represents only the match between the variables).

Sorry for the wall of text. Please let me know if something should be investigated or elaborated on more.

**Review by TorsteinHonsi — APPROVED** [review]
Thanks for the explanation, I understand you have considered all possibilities!
